From c66cb62b9f15e9a8e34ea006c227b06fe40a8de9 Mon Sep 17 00:00:00 2001
From: Michal Kazior <kazikcz@gmail.com>
Date: Sat, 19 Jan 2013 17:13:45 +0100
Subject: [PATCH] fix pasting from the X clipboard

Hello, 

Apparently once Shift is held XK_Y is reported instead of XK_y on my 
machine, so pasting from the X clipboard doesn't work. Attached patch 
fixes that. 

---
 dmenu.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dmenu.c b/dmenu.c
index 3962801..138dd48 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -274,7 +274,8 @@ keypress(XKeyEvent *ev) {
 			while(cursor > 0 && text[nextrune(-1)] != ' ')
 				insert(NULL, nextrune(-1) - cursor);
 			break;
-		case XK_y: /* paste selection */
+		case XK_y: /* fallthrough */
+		case XK_Y: /* paste selection */
 			XConvertSelection(dc->dpy, (ev->state & ShiftMask) ? clip : XA_PRIMARY,
 			                  utf8, utf8, win, CurrentTime);
 			return;
-- 
1.7.12.4

